plugins/terraform/data_source_layer0_deploy_test.go

```diff
@@ -0,0 +1,34 @@
+package main
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func TestDeployDataResourceSelectByQueryParams(t *testing.T) {
+	ctrl, mockClient, provider := setupUnitTest(t)
+	defer ctrl.Finish()
+
+	deployName := "deploy-name"
+	version := "1"
+
+	params := map[string]string{
+		"type":    "deploy",
+		"version": version,
+		"fuzz":    deployName,
+	}
+
+	mockClient.EXPECT().
+		SelectByQuery(params)
+
+	deployResource := provider.DataSourcesMap["layer0_deploy"]
+	d := schema.TestResourceDataRaw(t, deployResource.Schema, map[string]interface{}{})
+	d.Set("name", deployName)
+	d.Set("version", version)
+
+	if err := deployResource.Read(d, mockClient); err.Error() != fmt.Errorf("No entities found").Error() {
```

Comment:
The resource tests are a good point of reference. I think testing a service could look something like:
```
	ewts := []models.EntityWithTags{
		{EntityID: "id", Tags: []models.Tag{Key: "name", "value": "some_service"}},
	}

	mockClient.EXPECT().
		SelectByQuery(params).
		Return(ewts, nil)

       service := models.Service{ServiceID: "some_id", ServiceName: "some_service", ...}

	mockClient.EXPECT().
		GetService("id").
		Return(service, nil)

        // set d.RawData...

	if err := deployResource.Read(d, mockClient); err != nil {
		t.Fatal(err)
	}

      // maybe do some assertions that fields were set correctly
    assert.Equal(t, d.["service_name"], "some_service")
```